frontend/packages/dev-console/src/components/pipelines/pipeline-builder/utils.ts

```diff
@@ -93,6 +95,13 @@ const removeEmptyDefaultParams = (task: PipelineTask): PipelineTask => {
   return task;
 };
 
+export const removeEmptyDefaultFromPipelineParams = (parameters: PipelineParam[]) =>
```

Comment:
LGTM on implementation, but I think I am going to suggest 3 changes to help us out here.

1. Move this function `removeEmptyDefaultFromPipelineParams` out of the Pipeline Builder utils... maybe create a utility file in `frontend/packages/dev-console/src/components/pipelines/detail-page-tabs/`
2. Explicitly define the return type of `PipelineParam[]` so the caller knows they are giving you params, and getting back params in the same structure - maybe modified
3. Please write tests - this is a very testable function